From a2af6b8aaecd6d34603a81c9cb8867e340e9adb1 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair@alistair23.me>
Date: Mon, 18 Jan 2021 16:17:13 -0800
Subject: [PATCH 6/9] reMarkable: Don't load data from uboot.env

Ensure that we use out arguments intead of ones loaded from the FAT
partition.

Signed-off-by: Alistair Francis <alistair@alistair23.me>
---
 configs/zero-sugar_defconfig | 6 ------
 include/configs/zero-sugar.h | 7 +------
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/configs/zero-sugar_defconfig b/configs/zero-sugar_defconfig
index 4bb6841ea4..6105eeced9 100644
--- a/configs/zero-sugar_defconfig
+++ b/configs/zero-sugar_defconfig
@@ -16,7 +16,6 @@ CONFIG_SYS_CONSOLE_IS_IN_ENV=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="rM> "
 CONFIG_FASTBOOT=y
-CONFIG_FAT_WRITE=y
 CONFIG_FSL_FASTBOOT=y
 CONFIG_FASTBOOT_BUF_ADDR=0x83800000
 CONFIG_FASTBOOT_BUF_SIZE=0x40000000
@@ -45,14 +44,9 @@ CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
-CONFIG_CMD_SAVEENV=y
 CONFIG_EFI_PARTITION=y
 CONFIG_OF_CONTROL=y
-CONFIG_ENV_IS_IN_FAT=y
 # CONFIG_ENV_IS_IN_MMC is not set
-CONFIG_ENV_FAT_INTERFACE="mmc"
-CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
-CONFIG_ENV_FAT_ENV_FILE="uboot.env"
 CONFIG_DFU_MMC=y
 CONFIG_DFU_RAM=y
 CONFIG_DM_GPIO=y
diff --git a/include/configs/zero-sugar.h b/include/configs/zero-sugar.h
index 3384e0b09d..ee179e9d2e 100644
--- a/include/configs/zero-sugar.h
+++ b/include/configs/zero-sugar.h
@@ -152,7 +152,6 @@
 		"if test ${bootcount} -gt 10; then " \
 			"echo WARN: Failed too much, resetting bootcount and turning off; " \
 			"setenv bootcount 0; " \
-			"saveenv; " \
 			"poweroff; " \
 		"fi; " \
 		"setenv mmcpart ${fallback_partition}; " \
@@ -164,16 +163,12 @@
 
 /* Always try to boot from memory first, in case of USB download mode */
 #define CONFIG_BOOTCOMMAND \
-	"if test ! -e mmc 0:1 uboot.env; then " \
-		"saveenv; " \
-	"fi; " \
+	"echo Start; " \
 	"run memboot; " \
 	"run mmcargs; " \
 	"setenv mmcpart ${active_partition}; " \
 	"run mmcboot; " \
 	"echo WARN: unable to boot from either RAM or eMMC; " \
-	"setenv upgrade_available 1; " \
-	"saveenv; " \
 	"reset; "
 
 #define CONFIG_SYS_MEMTEST_START	0x80000000
-- 
2.29.2

